### PR TITLE
Added UnturnedVehicleReplacingBatteryEvent, UnturnedVehicleLockingEvent, UnturnedVehicleLockedEvent, and UnturnedVehicleUnlockingEvent

### PR DESCRIPTION
**Pull request by computeraims**
For your review: I'm not sure if I need to pass the group id in the Lock events since you can get the group ID from the instigator if you really need it. Thoughts?

### CODE REVIEW

**Comment by aviadmini** [comment]
You can get player's group ID, but I think vehicle might _technically_ be locked for custom group

**Comment by Trojaner** [comment]
I agree that group should stay

File: unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleLockedEvent.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using OpenMod.Unturned.Players;
+using Steamworks;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (IAmSilK) [inline unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleLockedEvent.cs:7]: Remove/sort usings

File: unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleUnlockingEvent.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using OpenMod.API.Eventing;
+using OpenMod.Unturned.Players;
+using Steamworks;
+
+namespace OpenMod.Unturned.Vehicles.Events
+{
+    public class UnturnedVehicleUnlockingEvent : UnturnedVehicleEvent, ICancellableEvent
```
Comment (IAmSilK) [inline unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleUnlockingEvent.cs:7]: Merge `UnturnedVehicleLockingEvent` and `UnturnedVehicleUnlockingEvent` into one event with a property of the locked state.

File: unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleLockedEvent.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace OpenMod.Unturned.Vehicles.Events
+{
+    public class UnturnedVehicleLockedEvent : UnturnedVehicleEvent
```
Comment (IAmSilK) [inline unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleLockedEvent.cs:11]: Add locked state property. This is available directly from the vehicle instance, but relevant info should be contained within the event's class.

File: unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs [hunk 4]
```diff
@@ -322,6 +388,71 @@ private static bool StealBattery(InteractableVehicle __instance, Player player)
...
+            {
+                InteractableVehicle vehicle = VehicleManager.findVehicleByNetInstanceID(instanceID);
+                if (vehicle is null)
+                    return;
+
+                Player nativePlayer = PlayerTool.getPlayer(owner);
+                if (nativePlayer is null)
+                    return;
+
+                if (locked)
```
Comment (IAmSilK) [inline unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs:453]: Refer to above, `locked` should be a parameter passed to `OnVehicleLocked` and should be called after being locked/unlocked

**Review by IAmSilK — CHANGES_REQUESTED** [review]
Overall, merge lock/unlock events. Consider changing names to describe both of these - perhaps something like `UnturnedVehicleLockChangingEvent`.

File: unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleLockedEvent.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿using OpenMod.Unturned.Players;
+using Steamworks;
+
+namespace OpenMod.Unturned.Vehicles.Events
+{
+    public class UnturnedVehicleLockedEvent : UnturnedVehicleEvent
```
Comment (IAmSilK) [inline unturned/OpenMod.Unturned/Vehicles/Events/UnturnedVehicleLockedEvent.cs:6]: Change to `UnturnedVehicleLockUpdatedEvent` for consistency.

File: unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs [hunk 6]
```diff
@@ -56,6 +59,9 @@ public override void Unsubscribe()
             OnVehicleExited -= Events_OnVehicleExited;
             OnVehicleSwapped -= Events_OnVehicleSwapped;
             OnVehicleStealBattery -= Events_OnVehicleStealingBattery;
+            OnVehicleReplacingBattery -= Events_OnVehicleReplacingBattery;
+            OnVehicleLockUpdating += Events_OnVehicleLockUpdating;
```
Comment (IAmSilK) [inline unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs:63]: Unsubscribe instead of a second subscription

**Review by IAmSilK — CHANGES_REQUESTED** [review]

File: unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs (hunk 6 above)
Comment (computeraims) [reply unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs:63]: Oops, ill try to me more careful with this.

**Review by IAmSilK — APPROVED** [review]
All looks good

**Pull request merged**